drivers: lte_lc: Ensure correct eDRX parameters are used

jtguggedal opened the pull request:
drivers: lte_lc: Ensure correct eDRX parameters are used

Check system mode and set ActT parameter accordingly
when requesting eDRX.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

---

drivers: lte_lc: Fix invalid eDRX default value for NB-IoT

This patch changes the default eDRX value from 1000 to 1001,
in order for it to be valid for both LTE-M and NB-IoT networks.
Ref 3GPP TS 24.008, subclause 10.5.5.32, table for S1 mode.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

jtguggedal commented:
@rlubos I think we should consider backporting to v1.1.0 branch. The issue is when the system mode fallback mechanism kicks, and for example switches from LTE-M to NB1. The eDRX parameter is still set by Kconfig, and will then be wrong, leading to eDRX not working for the fallback system mode.

jtguggedal commented:
@rlubos I'll create one

rlubos commented on drivers/lte_link_control/lte_lc.c:
```diff
@@ -453,9 +450,36 @@ int lte_lc_psm_get(int *tau, int *active_time)
+		break;
+	case LTE_LC_SYSTEM_MODE_NBIOT:
+	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
+		actt = 5;
```
Could we add symbols for these constants to the list above (somewhere near line 47)?

jtguggedal commented:
Updated with defines for the dynamically decided ActT parameter. 
Also, switched to using `lte_lc_edrx_req()` in the init function instead of AT command directly. NOTE: I'm not sure sending the eDRX request at that point has any effect, it could be too soon. Will check it and (re)move it if it needs to be.

jtguggedal commented:
> NOTE: I'm not sure sending the eDRX request at that point has any effect, it could be too soon. Will check it and (re)move it if it needs to be.

Had to add setting of system mode to the top of the init function as well. Needs to be set for eDRX to use correct parameter.

jtguggedal commented:
@rlubos I discovered another bug, and added a commit to fix it. You might want to revisit.
